R/experiment.R

```diff
@@ -1532,6 +1532,59 @@ Experiment <- R6::R6Class(
       saveRDS(self, file.path(save_dir, "experiment.rds"))
       invisible(self)
     },
+    export_visualizers = function(device = "pdf", width = "auto", height = "auto",
+                                ...) {
+      if (!rlang::is_installed("ggplot2")) {
+        stop("ggplot2 must be installed to export visualizers to image. Please install package via `install.packages('ggplot2')`.")
+      }
+      viz_list <- self$get_visualizers()
+      if (length(viz_list) == 0) {
+        return(invisible(self))
+      }
+      viz_results <- private$.get_cached_results("viz", verbose = 0)
+
+      if (!private$.has_vary_across()) {
```

Comment:
This is probably a bigger discussion, but I'm worried that the current way we create file paths via vary params names could become brittle, especially across runs / adding new params / transferring saved results from one filesystem to another (e.g., from a cluster to local machine). What do you think?

One question specific to this PR: this logic where we check for vary across and combine param names seems to be repeated in a few places. Should we extract it to a separate function, e.g. would it make sense to put it in `get_save_dir()` or a private version?

Reply:
I created a new private function `.get_vary_across_dir()` which returns the current `vary_across` directory.

If you're able to come up with an example where the file path changes, let me know. I think the file path should be consistent, as long as the simulation components are added in the same order. But yeah, this is something to think about more in the future.